core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/cassandra/MetricsServiceCassandra.java

```diff
@@ -701,9 +700,9 @@ private void dropKeyspace(String keyspace) {
                     return Futures.immediateFuture(results);
                 }
                 List<ResultSetFuture> insertFutures = new ArrayList<>();
-                tags.forEach((k, v) -> dataAccess.insertAvailabilityTag(k, v.orElse(""), data));
+                tags.forEach((k, v) -> dataAccess.insertAvailabilityTag(k, v, data));
                 for (Availability a : data) {
-                    insertFutures.add(dataAccess.updateDataWithTag(a, MetricUtils.flattenTags(tags)));
+                    insertFutures.add(dataAccess.updateDataWithTag(a, tags));
```

Comment:
To simplify the model and API, a tag is a tuple (key,value) with both required. We should provide richer query capabilities, but tag creation should be as simple as possible.